src/Compiler/Backend/GLSL/GLSLExtensionAgent.cpp

```diff
@@ -87,7 +87,7 @@ std::set<std::string> GLSLExtensionAgent::DetermineRequiredExtensions(
 
     if (separateShaders)
     {
-        if (shaderTarget != ShaderTarget::FragmentShader && shaderTarget != ShaderTarget::ComputeShader)
+        if (shaderTarget != ShaderTarget::ComputeShader)
```

Comment:
What's the reason for this change? As far as I remember, the original code for *separate shader objects* was introduced by @BearishSun so I'm not sure if this condition is sensible.

Reply:
You need to enable separate shaders extension in all shaders except compute shaders in order to work.

Also if you don't opengl will spam errors, and I tested it on my end.